### ISSUE TITLE
kornia.geometry.transform.warp_image_tps() returning black image

### ISSUE THREAD

**Issue by edoardoMalaspina** [kornia/kornia#2934]
### Describe the bug

warp_image_tps is returning a black image, both if tested with reasonable keypoints obtained with image registration frameworks and with random points. The  warp_points_tps() is working correctly and this method is exploited in the warp_image_tps  method, the problem seems to appear after the grid_sample() step in warp_image_tps() implementation, which returns always a torch tensor of the correct shape but filled with zeros.

### Reproduction steps

```bash
To reproduce the problem is sufficient to add some point to the example provided in the documentation 

points_src = torch.rand(1, 25, 2)
points_dst = torch.rand(1, 25, 2)
image = torch.rand(1, 3, 32, 32)
kernel_weights, affine_weights = get_tps_transform(points_dst, points_src)
warped_image = warp_image_tps(image, points_src, kernel_weights, affine_weights)
# warped image will be of the correct shape but will contain only zeros
```

### Expected behavior

It should output the warped image, in case of reasonable keypoints it should be a reasonable image

### Environment

```shell
- PyTorch Version (e.g., 1.0): torch 2.3.1+cu118
- OS (e.g., Linux): Same problem both on windows and ubuntu
- How you installed PyTorch (`conda`, `pip`, source): pip
```

**Comment by edgarriba** [comment]
@edoardoMalaspina could you provide a minimal colab reproducing the issue ?

**Comment by edoardoMalaspina** [comment]
Hello @edgarriba, first of all thank you for your answer! 
I'm attaching a notebook and a folder in which there are the keypoints that i computed with another framework to be used with it (the keypoints are reasonable because if I try to use them with the cv2 warp image it works as expected). 
In the notebook i use a simple image generated in the code just for simplicity but given the fact that the keypoints are reasonable it should apply a warping, instead it returns only 0s. Also you can see that i sliced the number of keypoints passed to TPS, because with the around 200 keypoints of the folder it was giving me an error ("the matrix was singular"), but even slicing it should apply some sort of warping instead of returning a complete black image. Thank you in advance and please let me know if you need a different kind of example in order to reproduce the problem!

[keypoints.zip](https://github.com/user-attachments/files/16086434/keypoints.zip)
[notebook.zip](https://github.com/user-attachments/files/16086522/notebook.zip)

**Comment by edoardoMalaspina** [comment]
I'm still investigating this issue and i noticed that if the values in output from the warp_points_tps are normalized (example warped /= torch.max(torch.abs(warped)) ) inside the warp_image_tps method the warped image in output is not completely black but is still outputting something unreasonable. Anyone had a similar issue?

**Comment by fawolfmann** [comment]
Im getting the same issue, we are thinking of moving out from kornia for this function, couldnt make it work

**Comment by edgarriba** [comment]
have you guys checked the tests to get a clue here ? https://github.com/kornia/kornia/blob/main/tests/geometry/transform/test_thin_plate_spline.py

I haven't used this function personally so I cannot be of much help. Maybe @catalys1 could guide since he's the original author of this code.

**Comment by catalys1** [comment]
`grid_sample()` requires coordinates normalized between -1 and 1: see [here](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html#:~:text=normalized%20by%20the%20input%20spatial%20dimensions.%20Therefore%2C%20it%20should%20have%20most%20values%20in%20the%20range%20of%20%5B%2D1%2C%201%5D). This is likely the issue based on the information provided here. Any points that back-warp to values outside of [-1, 1] will be filled with black pixels.

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
